sw/device/boot_rom/bootstrap.c

```diff
@@ -33,81 +39,97 @@ static int bootstrap_requested(void) {
   // Read pin
   uint32_t gpio_in;
   dif_gpio_all_read(&gpio, &gpio_in);
-  return !!(gpio_in & GPIO_BOOTSTRAP_BIT_MASK);
+  return gpio_in != 0 && GPIO_BOOTSTRAP_BIT_MASK != 0;
 }
 
-/* Erase all flash, and verify blank. */
+/**
+ * Erase all flash, and verify blank.
+ */
 static int erase_flash(void) {
-  if (flash_bank_erase(FLASH_BANK_0)) {
+  if (flash_bank_erase(FLASH_BANK_0) != 0) {
     return E_BS_ERASE;
   }
-  if (flash_bank_erase(FLASH_BANK_1)) {
+  if (flash_bank_erase(FLASH_BANK_1) != 0) {
     return E_BS_ERASE;
   }
-  if (!flash_check_empty()) {
+  if (flash_check_empty() == 0) {
     return E_BS_NOTEMPTY;
   }
+
   return 0;
 }
 
-/* Calculates SHA256 hash of received data and compares it against recieved
- * hash. Returns 0 if both hashes are identical. */
-static int check_frame_hash(const frame_t *f) {
-  static uint32_t hash[8];
-  uint32_t result = 0;
-  hw_SHA256_hash((uint8_t *)&f->hdr.frame_num, RAW_BUFFER_SIZE - 32,
-                 (uint8_t *)hash);
-  for (int i = 0; i < 8; ++i) {
-    result |= hash[i] ^ f->hdr.hash[i];
-  }
-  return result;
+/**
+ * Compares the SHA256 hash of the recieved data with the recieved hash.
+ *
+ * Returns true if the hashes match.
+ */
+static bool check_frame_hash(const spiflash_frame_t *frame) {
+  uint8_t hash[32];
```

Comment:
Could you drop a comment explaining what `32` means?

Reply:
They're both SHA2 hashes. But, sure.